Code/Engine/Foundation/Reflection/Implementation/PropertyAttributes.h

```diff
@@ -712,14 +712,58 @@ class EZ_FOUNDATION_DLL ezAutoGenVisScriptMsgSender : public ezPropertyAttribute
   EZ_ADD_DYNAMIC_REFLECTION(ezAutoGenVisScriptMsgSender, ezPropertyAttribute);
 };
 
-
 /// \brief Attribute for ezMessages to instruct the visual script framework to automatically generate a node for handling this type of
 /// message
 class EZ_FOUNDATION_DLL ezAutoGenVisScriptMsgHandler : public ezPropertyAttribute
 {
   EZ_ADD_DYNAMIC_REFLECTION(ezAutoGenVisScriptMsgHandler, ezPropertyAttribute);
 };
 
+/// \brief Attribute for ezMessages to instruct the visual script framework to automatically generate a node for handling this type of
+/// message
+class EZ_FOUNDATION_DLL ezScriptableFunctionAttribute : public ezPropertyAttribute
+{
+  EZ_ADD_DYNAMIC_REFLECTION(ezScriptableFunctionAttribute, ezPropertyAttribute);
+
+  ezScriptableFunctionAttribute(const char* szArg1 = nullptr, const char* szArg2 = nullptr, const char* szArg3 = nullptr, const char* szArg4 = nullptr, const char* szArg5 = nullptr, const char* szArg6 = nullptr)
+  {
+    m_sArg1 = szArg1;
+    m_sArg2 = szArg2;
+    m_sArg3 = szArg3;
+    m_sArg4 = szArg4;
+    m_sArg5 = szArg5;
+    m_sArg6 = szArg6;
+  }
+
+  const char* GetArgumentName(ezUInt32 index) const
+  {
+    switch (index)
```

Comment:
C has this great feature called arrays ;)

Reply:
not sure if that works with replication, since these attributes need to be sent across process boundaries

Reply:
Tried to use array properties, doesn't work with constant size arrays.